### PR TITLE
For #1539: Only allow delete when folders multi-selected 

### PR DESCRIPTION
**Pull request by colintheshots**
Tests were added along with the bookmark LibState changes.

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -695,4 +695,5 @@
     <string name="sign_out_disconnect">Disconnect</string>
     <!-- Option to cancel signing out shown in confirmation dialog to sign out of account -->
     <string name="sign_out_cancel">Cancel</string>
+    <string name="bookmark_cannot_edit_root">Cannot edit root folders</string>
```
Comment (boek) [inline app/src/main/res/values/strings.xml]: We're past string freeze for 1.1, so we'll have to wait until after we cut the release branch to land this.

**Comment by boek** [comment]
Waiting on string review

**Comment by colintheshots** [comment]
We have a string review and the PR is updated.

**Comment by colintheshots** [comment]
Updated this to work with the new LibStore changes and added a test.

**Pull request merged**